src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -353,38 +343,151 @@ def gz_to_pyarrow(
+            assert (
+                len(unique_list) == 1
+            ), f"{self.config_type} Table column {col} had {len(unique_list)} unique elements"
 
-            sort_log = ProcessLogger(
-                "pyarrow_sort_by", table_rows=table.num_rows
+    def sync_with_s3(self, part_folder: str) -> str:
+        """
+        check and see if UUID path exists on s3
+        """
+
+        local_folder = os.path.join(
+            self.tmp_folder,
+            part_folder,
+        )
+        if os.path.exists(local_folder):
+            return os.path.join(local_folder, os.listdir(local_folder)[0])
+
+        os.makedirs(local_folder, exist_ok=True)
+        bucket = os.environ["SPRINGBOARD_BUCKET"]
+        s3_files = file_list_from_s3(
+            bucket,
+            file_prefix=f"{part_folder}/",
+        )
+        if len(s3_files) == 1:
+            s3_path = s3_files[0].replace("s3://", "")
+            local_path = s3_path.replace(bucket, self.tmp_folder)
+            download_file(s3_path, local_path)
+            return local_path
+
+        return ""
+
+    # pylint: disable=R0914
+    def continuous_pq_update(self, table: pyarrow.Table) -> None:
+        """
+        Continuous updating of local parquet files that are synced with S3
+        """
+        log = ProcessLogger("continuous_pq_update")
+        log.log_start()
+        try:
+            self.verify_table(table)
+
+            year = pc.unique(table.column("year")).to_pylist()[0]
+            month = pc.unique(table.column("month")).to_pylist()[0]
+            day = pc.unique(table.column("day")).to_pylist()[0]
+            hour = pc.unique(table.column("hour")).to_pylist()[0]
+
+            part_folder = os.path.join(
+                DEFAULT_S3_PREFIX,
+                str(self.config_type),
+                f"year={year}",
+                f"month={month}",
+                f"day={day}",
+                f"hour={hour}",
             )
-            sort_log.log_start()
-            if self.detail.table_sort_order is not None:
-                table = table.sort_by(self.detail.table_sort_order)
-            sort_log.log_complete()
-
-            write_parquet_file(
-                table=table,
-                file_type=s3_prefix,
-                s3_dir=os.path.join(
-                    os.environ["SPRINGBOARD_BUCKET"],
-                    DEFAULT_S3_PREFIX,
-                    s3_prefix,
-                ),
-                partition_cols=["year", "month", "day", "hour"],
-                visitor_func=self.send_metadata,
+            local_path = self.sync_with_s3(part_folder)
+
+            log.add_metadata(part_folder=part_folder, local_path=local_path)
+
+            # no existing table file, write new file
+            if local_path == "":
+                local_path = os.path.join(
+                    self.tmp_folder,
+                    part_folder,
+                    f"{datetime(year, month, day, hour).isoformat()}.parquet",
+                )
```

Comment:
this has the building out of the filename we would look for on s3, with the s3 bucket instead of the temp folder obviously.